### ISSUE TITLE
Black screen on startup

### ISSUE THREAD

**Issue by vinukn335** [Codelessly/ResponsiveFramework#12]
I don't know this is bug or not

```dart
import 'package:flutter/material.dart';
import 'package:responsive_framework/responsive_framework.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
         builder: (context, widget) => ResponsiveWrapper.builder(
          BouncingScrollWrapper.builder(context, widget),
          maxWidth: 1200,
          minWidth: 450,
          defaultScale: true,
          breakpoints: [
            ResponsiveBreakpoint.resize(450, name: MOBILE),
            ResponsiveBreakpoint.autoScale(800, name: TABLET),
            ResponsiveBreakpoint.autoScale(1000, name: TABLET),
            ResponsiveBreakpoint.resize(1200, name: DESKTOP),
            ResponsiveBreakpoint.autoScale(2460, name: "4K"),
          ],
          background: Container(color: Color(0xFFF5F5F5))),
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
        // This makes the visual density adapt to the platform that you run
        // the app on. For desktop platforms, the controls will be smaller and
        // closer together (more dense) than on mobile platforms.
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/47185599/81787209-553f7100-951e-11ea-8432-fe960a94f6ff.gif)

**Comment by searchy2** [comment]
Thanks for trying out the Responsive Framework. The black screen you're seeing is likely the iOS native background while Flutter loads. You can set the background to white or a loading image. 

Please look through the following issues that are similar to yours:
https://github.com/flutter/flutter/issues/20827
https://flutter.dev/docs/development/ui/advanced/splash-screen

**Comment by searchy2** [comment]
I'll double check later today if you still experience the issue.

**Comment by vinukn335** [comment]
You are right. This bug related to flutter only. Your library is awesome!
Regards.

**Comment by vinukn335** [comment]
Without responsive_framework no black screen

![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/47185599/81941645-bc405100-9616-11ea-934b-2de2ce6f8cb4.gif)

**Comment by vinukn335** [comment]
Same problem happens in Android also

**Comment by searchy2** [comment]
This issue is pretty insane so thanks for bringing it to my attention. 

It's not just one black screen you're seeing, there's 3! black screens.

The first black screen appears when the emulator connects and flickers. This seems to be intermittent. This black screen doesn't seem to appear on an actual device or launching from the launcher on the emulator. 

The second black screen is because I made a mistake with the placeholder widget. I am returning a `SizedBox.shrink()` on the first frame instead of `Container(color: Colors.transparent)`. Due to the 1 frame layout calculation delay, you might need to set MaterialApp's `color` to `Colors.white`;

The third black screen is from the empty Scaffold background. Please set the Scaffold's `backgroundColor` to `Colors.white`. 

```dart
import 'package:flutter/material.dart';
import 'package:responsive_framework/responsive_framework.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      builder: (context, widget) => ResponsiveWrapper.builder(
          BouncingScrollWrapper.builder(context, widget),
          maxWidth: 1200,
          minWidth: 450,
          defaultScale: true,
          breakpoints: [
            ResponsiveBreakpoint.resize(450, name: MOBILE),
            ResponsiveBreakpoint.autoScale(800, name: TABLET),
            ResponsiveBreakpoint.autoScale(1000, name: TABLET),
            ResponsiveBreakpoint.resize(1200, name: DESKTOP),
            ResponsiveBreakpoint.autoScale(2460, name: "4K"),
          ],
          background: Container(color: Color(0xFFF5F5F5))),
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
      color: Colors.white,
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      appBar: AppBar(title: Text('Flutter App')),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ),
    );
  }
}
```
![ezgif com-video-to-gif (3)](https://user-images.githubusercontent.com/19896174/81997076-5b177e00-9614-11ea-8f2f-0c2977589f66.gif)

I'll release a fix this weekend for this bug.

**Comment by searchy2** [comment]
Ooops, let's keep this open until the fix is merged.

**Comment by searchy2** [comment]
Sorry for the delay. There appears to be a behavior change with how Container dimensions are calculated. Holding off on pushing a new release until more testing and unit tests are done. 

For now, you can use the fix on the latest branch with:
```
responsive_framework:
    git:
      url: https://github.com/Codelessly/ResponsiveFramework.git
      ref: master
```

**Comment by viveky259259** [comment]
Is it solved ?

**Comment by searchy2** [comment]
Sorry, the fix is not yet released. I'll have more time soon to devote to the Responsive Framework soon.

**Comment by viveky259259** [comment]
Hi, I found solution to my problem. I was using an expanded on a place and because of that, I was getting this error.
 
After removing the Expanded Widget I was able to get rid of this problem

**Comment by searchy2** [comment]
The issue is now fixed and available in v0.0.8.
There is no need to add any MaterialApp or Scaffold backdrop color anymore. 
Please note, when an app is launched the first time in the simulator, there can still be a black screen due to startup timing.
